libpkg/pkgdb.c

```diff
@@ -2880,25 +2880,24 @@ static int
 pkgdb_try_lock(struct pkgdb *db, const char *lock_sql, pkgdb_lock_t type,
 		bool upgrade)
 {
-	unsigned int tries = 0;
+	int64_t tries_remaining = 1;
 	struct timespec ts;
 	int ret = EPKG_END;
-	const pkg_object *timeout, *max_tries;
+	const pkg_object *timeout, *max_retries;
 	double num_timeout = 1.0;
-	int64_t num_maxtries = 1;
 	const char reset_lock_sql[] = ""
 			"DELETE FROM pkg_lock; INSERT INTO pkg_lock VALUES (0,0,0);";
 
 
 	timeout = pkg_config_get("LOCK_WAIT");
-	max_tries = pkg_config_get("LOCK_RETRIES");
+	max_retries = pkg_config_get("LOCK_RETRIES");
 
 	if (timeout)
 		num_timeout = pkg_object_int(timeout);
-	if (max_tries)
-		num_maxtries = pkg_object_int(max_tries);
+	if (max_retries)
+		tries_remaining = 1 + pkg_object_int(max_retries);
 
-	while (tries <= num_maxtries) {
+	while (tries_remaining == -1 || tries_remaining-- > 0) {
```

Comment:
As noted on IRC: if LOCK_RETRIES == -1, tries_remaining = 0 going into this. tries_remaining != -1 and !(tries_remaining-- > 0)

Reply:
Thanks for catching! I think I've fixed this now.